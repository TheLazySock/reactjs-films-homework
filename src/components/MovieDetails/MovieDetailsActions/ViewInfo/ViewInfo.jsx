import React from 'react';
import PropTypes from 'prop-types';
import style from './ViewInfo.scss';

import Button from '../../../Button';

class ViewInfo extends React.Component {
  state = {
    isInfoShown: false,
  }

  handleShowInfo = () => {
    this.setState(prevState => ({ isInfoShown: !prevState.isInfoShown }));
  }

  render() {
    const { info } = this.props;
    const { isInfoShown } = this.state;

    return (
      <div className={style.viewInfoContainer}>
        {
          isInfoShown && (
            <div>
              <p className={style.modalInfo}>{info}</p>
            </div>
          )
        }
        <Button buttonClass="viewInfoBtn" onClick={this.handleShowInfo} title="View Info" />
      </div>
    );
  }
}

ViewInfo.propTypes = {
  info: PropTypes.string.isRequired,
};

export default ViewInfo;
