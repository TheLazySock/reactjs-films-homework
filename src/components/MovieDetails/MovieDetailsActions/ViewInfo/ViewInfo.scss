.viewInfoContainer {
  position: relative;
}

.viewInfoBtn {
  padding: 7px 15px;
  border: 1px solid white;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  font-size: 14px;
  background-color: #000;

  &:hover {
    background-color: #313131;
  }

  &:active {
    background-color: #000;
  }
}

.modalInfo {
  position: absolute;
  margin: 0;
  padding: 10px 15px;
  width: 210px;
  bottom: 60px;
  right: -70px;
  font-size: 12px;
  line-height: 20px;
  letter-spacing: 0.5px;
  background-color: rgba(0, 0, 0, 0.5);

  &::before {
    position: absolute;
    content: '';
    left: 50%;
    margin-left: -15px;
    bottom: -15px;
    width: 0;
    height: 0;
    border-left: 15px solid transparent;
    border-right: 15px solid transparent;
    border-top: 15px solid rgba(0, 0, 0, 0.5);
  }
}
