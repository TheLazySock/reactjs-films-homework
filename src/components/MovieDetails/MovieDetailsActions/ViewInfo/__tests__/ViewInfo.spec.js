import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';

import ViewInfo from '../index';

describe('ViewInfo', () => {
  it('renders properly', () => {
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<ViewInfo onClick={jest.fn()} info="info" />);
    const output = shallowRenderer.getRenderOutput();
    expect(output.type).toBe('div');
    expect(output).toMatchSnapshot();
  });

  it('show description on button click', () => {
    const output = renderer.create(<ViewInfo videoUrl="https://" info="info" />);
    output.root.findByProps({ className: 'viewInfoBtn' }).props.onClick();
    expect(output).toMatchSnapshot();
  });
});
