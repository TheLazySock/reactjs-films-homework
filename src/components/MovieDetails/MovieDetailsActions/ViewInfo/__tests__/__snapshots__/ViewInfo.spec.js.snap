// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewInfo renders properly 1`] = `
<div
  className="viewInfoContainer"
>
  <Button
    buttonClass="viewInfoBtn"
    onClick={[Function]}
    title="View Info"
  />
</div>
`;

exports[`ViewInfo show description on button click 1`] = `
<div
  className="viewInfoContainer"
>
  <div>
    <p
      className="modalInfo"
    >
      info
    </p>
  </div>
  <button
    className="viewInfoBtn"
    onClick={[Function]}
    type="button"
  >
    View Info
  </button>
</div>
`;
